sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala

```diff
@@ -160,9 +160,20 @@ case class GenerateExec(
 
     // Generate looping variables.
     val index = ctx.freshName("index")
+    val numElements = ctx.freshName("numElements")
+
+    // In case of outer=true we need to make sure the loop is executed at-least once when the
+    // array/map contains no input.
+    // generateOuter is an int. it is set to 1 iff outer is true and the input is empty or null.
+    val generateOuter = ctx.freshName("generateOuter")
+    val isOuter = if (outer) {
```

Comment:
How about we generate the expressions for the `generateOuter` variable here instead? It is always `0` when `outer == false` and `$numElements == 0 ? 1 : 0` when it is true. This increases the probability that the generated code gets eliminated at compile time.